documentation/0013-node-output-to-cli/index.md

```diff
@@ -0,0 +1,191 @@
+```
+
+and Node will print both.
+
+We can also format pretty phrases by passing variables and a format specifier.
+
+For example:
+
+```js
+console.log('My %s has %d years', 'cat', 2)
+```
+
+- `%s` format a variable as a string
+- `%d` or `%i` format a variable as an integer
+- `%f` format a variable as a floating point number
+- `%O` used to print an object representation
+
+Example:
+
+```js
+console.log('%O', Number)
+```
+
+## Clear the console
+
+`console.clear()` clears the console (the behavior might depend on the console used)
+
+## Counting elements
+
+`console.count()` is a handy method.
+
+Take this code:
+
+```js
+const x = 1
+const y = 2
+const z = 3
+console.count(
+  'The value of x is ' + x + ' and has been checked .. how many times?'
+)
+console.count(
+  'The value of x is ' + x + ' and has been checked .. how many times?'
+)
+console.count(
+  'The value of y is ' + y + ' and has been checked .. how many times?'
+)
+```
+
+What happens is that count will count the number of times a string is printed, and print the count next to it:
+
+You can just count apples and oranges:
+
+```js
+const oranges = ['orange', 'orange']
+const apples = ['just one apple']
+oranges.forEach(fruit => {
+  console.count(fruit)
+})
+apples.forEach(fruit => {
+  console.count(fruit)
+})
+```
+
+## Print the stack trace
+
+There might be cases where it's useful to print the call stack trace of a function, maybe to answer the question _how did you reach that part of the code?_
+
+You can do so using `console.trace()`:
+
+```js
+const function2 = () => console.trace()
+const function1 = () => function2()
+function1()
+```
+
+This will print the stack trace. This is what's printed if we try this in the Node REPL:
+
+```txt
+Trace
+    at function2 (repl:1:33)
+    at function1 (repl:1:25)
+    at repl:1:1
+    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
+    at REPLServer.defaultEval (repl.js:239:29)
+    at bound (domain.js:301:14)
+    at REPLServer.runBound [as eval] (domain.js:314:12)
+    at REPLServer.onLine (repl.js:440:10)
+    at emitOne (events.js:120:20)
+    at REPLServer.emit (events.js:210:7)
+```
+
+## Calculate the time spent
+
+You can easily calculate how much time a function takes to run, using `time()` and `timeEnd()`
+
+```js
+const doSomething = () => console.log('test')
+const measureDoingSomething = () => {
+  console.time('doSomething()')
+  //do something, and measure the time it takes
+  doSomething()
+  console.timeEnd('doSomething()')
+}
+measureDoingSomething()
+```
+
+## stdout and stderr
+
+As we saw console.log is great for printing messages in the Console. This is what's called the standard output, or `stdout`.
+
+`console.error` prints to the `stderr` stream.
+
+It will not appear in the console, but it will appear in the error log.
+
+## Color the output
+
+You can color the output of your text in the console by using escape sequences. An escape sequence is a set of characters that identifies a color.
```

Comment:
A link to [ANSI color table](https://gist.github.com/chrisopedia/8754917) would probably make this much more decipherable IMO

Reply:
Good idea!